runtime/doc/api.txt

```diff
@@ -3108,6 +3108,204 @@ nvim_tabpage_set_var({tabpage}, {name}, {value})
                     {value}    Variable value
 
 
+==============================================================================
+Autocmd Functions                                                *api-autocmd*
+
+nvim_create_augroup({*opts})                           *nvim_create_augroup()*
+                Create or get an autocommand group |autocmd-groups|.
+
+                To get an existing augroup id, do: >
+                    local id = vim.api.nvim_create_augroup({
+                      name = "MyGroup",
+                      clear = false
+                    });
+<
+
+                Parameters: ~
+                    {opts}  Dictionary Parameters
+                            • name: (string) The name of the augroup
+                            • clear: (bool) optional, defaults to true. Clear
+                              existing commands if the group already exists
+                              |autocmd-groups|.
+
+                Return: ~
+                    The id of the created augroup.
+
+                See also: ~
+                    |autocmd-groups|
+
+nvim_create_autocmd({*opts})                           *nvim_create_autocmd()*
+                Define a new autocommand |autocmd|.
+
+                The API allows for two types of actions to be executed
+                (mutually exclusive) when the autocommand triggers: a callback
+                function (Lua or Vimscript), or a command (like regular
+                autocommands).
+
+                Example using callback: >
+                    -- Lua function
+                    local myluafun = function() print("This buffer enters") end
+
+                    -- Vimscript function name (as a string)
+                    local myvimfun = "g:MyVimFunction"
+
+                    vim.api.nvim_create_autocmd({
+                      event = {"BufEnter", "BufWinEnter"},
+                      pattern = {"*.c", "*.h"},
+                      callback = myluafun,  -- Or myvimfun
+                    })
+<
+
+                Example using command: >
+                    vim.api.nvim_create_autocmd({
+                      event = {"BufEnter", "BufWinEnter"},
+                      pattern = {"*.c", "*.h"},
+                      command = "echo 'Entering a C or C++ file'",
+                    })
+<
+
+                Example values for patterns: >
+                  pattern = "*.py,*.pyi"
+                  pattern = "*.py"
+                  pattern = {"*.py"}
+                  pattern = { "*.py", "*.pyi" }
+                  -- NOT supported
+                  pattern = {"*.py,*.pyi"}
+<
+
+                Examples values for events: >
+                  event = "FileType,CursorHold"
+                  event = "BufPreWrite"
+                  event = {"BufPostWrite"}
```

Comment:
I would remove that example, too -- basically, just show "string for single event, table for multiple events" (and the leave trivial corollary "one is a special case of multiple" as an exercise to the reader).